File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityRelativeMovePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record EntityRelativeMovePacket(int entityId, byte deltaX, byte deltaY, byte deltaZ)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(deltaX);
        os.writeByte(deltaY);
        os.writeByte(deltaZ);
    }

    @Override
    public byte getId() {
        return 0x15;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityRelativeMovePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityRelativeMovePacket(val entityId: Int, val deltaX: Byte, val deltaY: Byte, val deltaZ: Byte) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(deltaX, deltaY, deltaZ)
    }

    override val id: Byte
        get() = 0x15
}
